### PR TITLE
10-14 Punchlist

### PR DESCRIPTION
**Pull request by AmbientOSX**
See commits.

@kadst43 @BrettASwift

### CODE REVIEW

File: source/view/blocks.js [hunk 1]
```diff
@@ -4867,8 +4852,8 @@ Blockly.Blocks['init_nano_construction'] = {
   init: function() {
     this.appendDummyInput()
         .appendField("Start Construction");
-    this.setPreviousStatement(true);
-    this.setNextStatement(true);
+    this.setPreviousStatement(true,'Normal');
+    this.setNextStatement(true,'Normal');
```
Comment (kadst43) [inline source/view/blocks.js]: I'm curious... what does this do?

**Comment by kadst43** [comment]
Cosmetically, looks much better.

This doesn't seem to address the issue where M4T14 doesn't succeed when it should - I assume that you expected that.

I'm seeing a lot of issues (playing through M4) where the START button is initially greyed, even though I have a good program.  I disconnect and reconnect a block, and the button comes un-gray.  I saw that on very nearly every scenario.  I think it's specific to this branch... maybe something to do with the RESET changes?

File: source/view/blocks.js (hunk 1 above)
Comment (AmbientOSX) [inline source/view/blocks.js]: It makes it so these blocks can only connect to blocks with connection statements set to nothing (this.setPreviousStatement(true)) or blocks that have the same string (can also be an array of types)... so a this.setNextStatement(true,'Normal') will only allow blocks with their setPreviousStatement set to 'Normal' or nothing.

**Comment by kadst43** [comment]
:+1:

**Comment by kadst43** [comment]
(you probably want to rebase before merge)

**Comment by AmbientOSX** [comment]
@kadst43 rebased - take a quick look again and then I'll merge..

**Comment by kadst43** [comment]
:+1:

**Pull request merged**